### DISCUSSION TITLE
表格的字号问题，应当在文档中加以说明

### DISCUSSION THREAD

**Discussion by FeynmanCheng** [nju-lug/NJUThesis#94] [Ideas]
按照教程内给出的表格代码示例，表格的内容字号和正文是一样的，都是小四号。
```tex
\begin{table}[htbp]
\centering
\caption{我的表}
\label{tab:testtab}
\begin{tabular}{ccc}
\toprule
OS & TeX & 测试情况\\
\midrule
南大TeX & Overleaf & √ \\
\bottomrule
\end{tabular}
\end{table}
```

自己使用时，添加了`\small`可使表格字体为五号。
建议在教学文档中说明字号的问题，方便像我一样的小白使用

**Comment by atxy-blip** [comment]
作为制作者，我个人希望用户不去管字号设置。模板之所以是模板，就在于提供了一个统一的解决方案，让用户不必手动指定格式。如果表格超出了文字宽度，或许应当考虑是不是内容太多、应当放在附录中占整页作为补充材料。

不论如何，稍后手册中还是会加一点对于字体格式的说明。

**Reply by FeynmanCheng** [reply]
按照本科生毕设的格式规定，表格字体应当是五号。

`表的题目在表的上方，图的题目在图的下方。图表的题目及内容的字体采用五号宋体。表中内容采用单倍行距。`

既然是模板，那么应当符合规范才是。我的意思是，按照教程里给出的使用表格代码，绘制出的表格内容是小四号字。我希望教程里加以提示，或是制作者添加一个全局表格字号设置。

**Reply by atxy-blip** [reply]
我爬，马上就加

**Reply by zepinglee** [reply]
可以修改 `\@floatboxreset` 设置表格内容的默认字号，参考 <https://www.zhihu.com/question/366803177/answer/977853129>。

**Reply by atxy-blip** [reply]
我选择了 `\BeforeBeginEnvironment { tabular } { \centering \small }` （逃

**Reply by note286** [reply]
[IEEEtran.cls](http://mirrors.ctan.org/macros/latex/contrib/IEEEtran/IEEEtran.cls)中对此有相关实现，直接重定义浮动体的好处是可以支持任意表格，只要是在`table`浮动体内的都会自动修改字号。

```latex
\def\table{\def\@floatboxreset{\reset@font\footnotesize\@setminipage}\@float{table}}
\def\endtable{\end@float}
```

只对`tabular`加钩子会不会使得其他表格环境无法生效，亦或者是一般其他表格环境内部均加载`tabular`环境。

**Reply by note286** [reply]
@atxy-blip 我在[xduts](https://github.com/note286/xduts/blob/e3097709684d8906218acbe643c5021eddf858a5/xduts.dtx#L3260)对此做了适配，应该能覆盖目前主流表格了，浮动的由`table`浮动体搞定，不浮动的跨页表格提供了`longtable`环境、`tabularray`宏包的`tblr`和`longtblr`环境适配，应该都够用了，可以搬去用。

**Reply by atxy-blip** [reply]
我看到 <https://github.com/note286/xduts/commit/3394ca05922e5bea73146f004147930d956bd183> 了。最近比较忙，暑假再做一次大范围修改。感谢~

**Comment by xiaoqiangch** [comment]
我搞了几个小时 adjust box ,\small \tiny \footnotesize 全都是试过了，都不管用，文档真的该明确说明一下，太折腾了，我觉得设计原则是：
**不需要用户管字体，但是，用户想要管字体的时候能够管的掉了**

**Reply by atxy-blip** [reply]
用户文档的编写原则是*尽可能简明*地描述所有公开接口，而在*实现细节*部分字面意思地解释做了什么。对于表格字号的问题，手册里面显然是有介绍的，请看下图。打开 PDF 阅读器然后按 `Ctrl`+`F` 搜索“表格”二字就能找到。在“版本历史”一节我们也可以发现在 v0.16 设置了表格字号为五号，变更发生在手册 68 页。

我会在日后有空时考虑要不要加入这个用户接口。无论如何，如果对模板设计原则*有异议*，随时可以 Fork 一份本项目并在 LPPL 1.3c 许可证的条款下自行改动，届时欢迎通过 Pull requests 贡献。

![image](https://github.com/nju-lug/NJUThesis/assets/63113546/cab61c67-ff65-41e1-a88e-fb5458fec30b)

**Reply by xhgt** [reply]
当表格很长时，一般会使用longtable包。我发现模板正常表格的字号是五号，但longtable的默认字号明显更大。现在看来，应该是需要用\BeforeBeginEnvironment{longtable}{\zihao{5}}调整一下就可以了。建议在下一个版本中把longtable内置到模板中，方便我这样的小白。
